File: backend/src/main/java/com/mtumer/services/UserRoleService.java
package com.mtumer.services;

import java.util.List;
import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.mtumer.entity.UserRole;
import com.mtumer.repo.UserRoleRepo;

@Service
public class UserRoleService {

	
	@Autowired
	UserRoleRepo userRoleRepo;
	
	public List<UserRole> getAllUserRoles(){
		return userRoleRepo.findAll();
		
	}
	
	public UserRole createUserRole(UserRole userRole) {
		UserRole newUserRole=new UserRole();
		newUserRole.setRole(userRole.getRole());
		newUserRole.setUser(userRole.getUser());
		newUserRole = userRoleRepo.save(newUserRole);
		return newUserRole;
	}
	
	public Optional<UserRole> getUserRoleById(Long userRoleId){
		return userRoleRepo.findById(userRoleId);
	}
	
	public void update(UserRole userRole) {
		userRoleRepo.saveAndFlush(userRole);
		
	}
	
	public void deleteUserRole(Long userRole) {
		userRoleRepo.deleteById(userRole);
	}
}

File: frontend/src/services/UserRoleService.js
import axios from "axios";

const PRODUCT_API_BASE_URL = "https://backendecommerce.azurewebsites.net/user_role";


class UserRoleService {

    findAllUserRole() {
        return axios.get(PRODUCT_API_BASE_URL + "/show");
    }
    getUserRoleById(id){
        return axios.get(`https://backendecommerce.azurewebsites.net/user_role/show/${id}`);
    }
    delete(id){
        return axios.delete(`https://backendecommerce.azurewebsites.net/user_role/delete/${id}`);
    }
    update(id,userRole){
        return axios.put(`https://backendecommerce.azurewebsites.net/user_role/update/${id}`, userRole);
    }
    createUserRole(userRole){
        return axios.post(`https://backendecommerce.azurewebsites.net/user_role/save`, userRole);
    }


}



export default new UserRoleService();
